tests/unit/s2n_openssl_x509_test.c

```diff
@@ -192,15 +210,16 @@ int main(int argc, char **argv)
         /* assert that cert info matches expected values */
         EXPECT_EQUAL(leaf_info.signature_nid, test_cases[i].expected_signature_nid);
         EXPECT_EQUAL(leaf_info.signature_digest_nid, test_cases[i].expected_digest_nid);
+        EXPECT_EQUAL(leaf_info.public_key_nid, test_cases[i].expected_public_key_nid);
+        EXPECT_EQUAL(leaf_info.public_key_bits, test_cases[i].expected_public_key_bits);
         EXPECT_EQUAL(leaf_info.self_signed, false);
 
         /* leaf and intermediate should have the same infos */
         EXPECT_EQUAL(memcmp(&leaf_info, &intermediate_info, sizeof(struct s2n_cert_info)), 0);
 
-        /* root should be self-signed */
-        EXPECT_EQUAL(root_info.signature_nid, test_cases[i].expected_signature_nid);
-        EXPECT_EQUAL(root_info.signature_digest_nid, test_cases[i].expected_digest_nid);
-        EXPECT_EQUAL(root_info.self_signed, true);
```

Comment:
Nope! It's still there since we set `leaf_info.self_signed = true` before the comparison. But I refactored this test to use an explicit `expected_info` struct to hopefully make this easier to follow.